R/estimate_ascertainment.R

```diff
@@ -0,0 +1,144 @@
+#' Estimate the ascertainment ratio of a disease
+#'
+#' @description Estimates the proportion of cases or infections that have been
+#' ascertained, given a time-series of cases and deaths, a delay distribution
+#' and a baseline severity estimate. The resulting ascertainment estimate is
+#' calculated as the ratio of the baseline severity estimate, which is assumed
+#' to be the 'true' disease severity, and the delay-adjusted severity estimate.
+#'
+#' @inheritParams cfr_time_varying
+#' @inheritParams cfr_static
+#'
+#' @param type A string, either `"static"` or `"varying"` which determines
+#' whether [cfr_static()] or [cfr_time_varying()] is used to calculate
+#' the resulting ascertainment ratio. Defaults to `"static"` if this argument is
+#' missing.
+#' @param severity_baseline A single number in the range 0.0 -- 1.0 for the
+#' assumed true baseline severity estimate used to estimate the overall
+#' ascertainment ratio. Missing by default, which causes the function to error;
+#' must be supplied by the user.
+#' @param max_date A `Date` representing a user supplied maximum date, up to
+#' which the time-varying severity estimate will be calculated. Useful in the
+#' case of long time-series, where the user wishes to focus on a specific
+#' time-period. See [as.Date()] for converting a string to a `Date`.
+#'
+#' @return A `<data.frame>` containing the maximum likelihood estimate estimate
+#' and 95% confidence interval of the corrected severity, named
+#' "ascertainment_mean" (for the central estimate), and "ascertainment_low" and
+#' "ascertainment_high" for the lower and upper interval limits.
+#' @export
+#'
+#' @examples
+#' # get data pre-loaded with the package
+#' data("covid_data")
+#' df_covid_uk <- covid_data[covid_data$country == "United Kingdom", ]
+#'
+#' df_covid_uk_subset <- subset(df_covid_uk, date <= "2020-05-31")
+#'
+#' # use a severity baseline of 1.4% (0.014) taken from Verity et al. (2020)
+#' # Lancet Infectious Diseases: 10.1016/S1473-3099(20)30243-7
+#'
+#' # use onset-to-death distribution from Linton et al. (2020)
+#' # J. Clinical Medicine: 10.3390/jcm9020538
+#'
+#' estimate_ascertainment(
+#'   data = df_covid_uk,
+#'   delay_density = function(x) dlnorm(x, meanlog = 2.577, sdlog = 0.440),
+#'   type = "varying",
+#'   severity_baseline = 0.014,
+#'   burn_in = 7L,
+#'   max_date = as.Date("2020-06-30")
+#' )
+#'
+estimate_ascertainment <- function(data,
+                                   delay_density = NULL,
+                                   type = c("static", "varying"),
```

Comment:
from a user perspective I wound find it a bit confusing that this is an argument here but in two different functions for cfr. Perhaps should use the same approach for both?

Reply:
Thanks - I can split the function and will continue this discussion in the main thread.

Reply:
Per main thread (and your later clarification) I will leave this fn as is for now.